### DISCUSSION TITLE
ISY as Hub?

### DISCUSSION THREAD

**Discussion by mtwunch** [kuestess/homebridge-platform-insteonlocal#322] [Q&A]
Does this plug-in support the ISY as the Insteon hub?  If so, any chance you have a config snippet?

**Comment by kuestess** [comment]
@mtwunch This plugin doesn't support the ISY, however I think there are other plugins that do.  You can use this plugin to interface directly with the PLM (without the ISY).

**Comment by mtwunch** [comment]
@kuestess Thank you!  I am using Homebridge to slowly move away from my Insteon network to matter/threads controlled by HomeKit.  I was attracted to this plugin because it supports the Insteon battery powered devices.  I will have a look for a plugin that supports the ISY.

**Comment by mtwunch** [comment]
@kuestess How do you connect to the PLM directly?  The connection between the PLM and ISY is a serial connection.

**Comment by mtwunch** [comment]
@kuestess I think I posted too hastily.  "Direct" connection to the PLM is possible with a serial device server.  Is this correct?

**Reply by kuestess** [reply]
@mtwunch Yes, you can connect directly to a serial PLM and this is how I have mine setup now.  Here's a sample portion of the config that you'll need:

```
        {
            "name": "Insteon Local Platform",
            "host": "/dev/ttyUSB0",
            "model": "PLM",
            "refresh": "0",
            "server_port": "3001",
            "keepAlive": "0",
            "devices": []
        }
```

You'll need to define devices, and change to the correct port, etc. but this should get you connected over serial to the PLM.

**Comment by mtwunch** [comment]
@kuestess I have the PLM connected to a serial interface server.  I am getting an error "serial is Unsupported on this platform".  The platform config snippet is:
{
    "platform": "InsteonLocal",
    "name": "InsteonLocal",
    "host": "192.168.68.125",
    "port": "4001",
    "model": "PLM",
    "refresh": "0",
    "keepAlive": "0",
    "devices": []
}
Any ideas as to what is incorrect?  
Thanks!
Mike

**Comment by kuestess** [comment]
@mtwunch Could be a couple of things.  First, you'll need to change the `host:` value to the path to your serial port where the PLM is plugged in rather than an IP address.  See my example above - if you're on Mac/Linux it's probably something like `/dev/ttyUSB0`.  Try that first.  If you still get the 'serial not supported message', it's probably a (not uncommon) issue with the `serialport` package.  You can likely fix that with `npm i -g serialport` in the terminal.

**Comment by mtwunch** [comment]
@kuestess I can configure the Moxa serial interface server to Real Comm mode.  To communicate, one also needs to install the Moxa drivers on a host, linux, MacOS, etc.  I am running Homebridge on a Synology NAS system.  What is not clear to me is how Homebridge knows to communicate with the host running those drivers.  Perhaps I am not understanding how TTY works.  Would you please elaborate on how you have your system setup?  Or if you know how others running Homebridge on a Synology NAS are setup that would be super beneficial.  Thank you!

**Comment by kuestess** [comment]
@mtwunch I'm not familiar with the Moxa interface - quick google search looks like its a serial to IP interface....but not sure you need that.  Easiest way, and how I have it set up, is to plug the PLM directly into USB on whatever your host machine is (Synology in your case, mine is an old Mac mini running arch linux) that you're running homebridge on.  Since you're on a Synology, I assume that you're running home bridge in Docker - if so, you'll need to map the specific serial port to the Docker container to get it to work.  If you really want to, you can get it working over a network connection with software - see here: [https://github.com/kuestess/homebridge-platform-insteonlocal/issues/122#issuecomment-732335760](https://github.com/kuestess/homebridge-platform-insteonlocal/issues/122#issuecomment-732335760) but I'd make sure that it works via direct USB connection first.

**Comment by mtwunch** [comment]
@kuestess Insteon makes two PLM models, one is a USB interface, the other is a serial interface.  I have the serial interface version because that is what the ISY uses to communicate with the PLM.  Which model are you using?  

Also, there are cables that map a serial connection into USB.  The cables have a chip in them that creates a virtual serial port.  These usually require a driver on the host, however.

The Moxa is a serial interface server.  It can be configured to map TCP into a serial data stream.  It can also be configured as a Real Comm device.  With drivers on a host, the unit looks like a serial port to the host.  

I do appreciate the time you are taking to help me figure this out!

**Comment by kuestess** [comment]
@mtwunch Happy to help!  I've got the USB PLM, so easy to plug directly into the host.  Moxa sounds like a cool device - in the end, you'll need to emulate a serial port on the Synology, so you'll probably need to install Moxa drivers on the Synology if they exist.  If not, you could probably get it to work via Docker but it will likely be painful on Synology (just because of how Synology deals with Docker permissions).  If you have another Linux/Mac machine around, that may be easier - even a Raspberry Pi will work.

**Comment by mtwunch** [comment]
@kuestess This is quite helpful.  With your setup, how do you make changes to your Insteon network, e.g. adding a device, creating a scene, or replacing a failed device?  How would you replace the PLM should it fail?  How would you restore the links to the new PLM?

**Comment by kuestess** [comment]
@mtwunch The plugin has a web UI as well that you can use to manage the Insteon network, kinda like the old HouseLinc software.  It's likely not as full-featured as what an ISY can do, but it can do all that you mention above except migrating links to a new PLM.  The groundwork is there for that feature, but I have not had time to implement it.

**Comment by mtwunch** [comment]
@kuestess Thank you!  Right now I am thinking that I will install Home Assistant on Raspberry Pi along with Insteon and Homebridge integrations.  I'll connect the PLM to the Pi with a serial to USB cable.  On Homebridge, I'll use the insteonlocal plugin.  This setup should allow me to manage the Insteon network and provide a bridge between old and new, with old being the Insteon devices and new being Matter/Thread devices in Homekit.  The insteonlocal plugin supports battery powered remotes and leak sensors (I have both).  None of the other plugins I looked at supported both of those battery powered devices.  Insteon is old technology, so as those devices die, I'll decide whether to replace them with Insteon devices or Matter/Thread devices.  Even though Insteon is old, it seems like some of their devices are still the best available.  Over time, if/when Matter/Thread takes off, the situation may be different.

**Reply by kuestess** [reply]
@mtwunch Ironically, the email from Insteon today has refurbished HubPro's for $25 [https://shop.insteon.com/products/hub-pro-refurbished](https://shop.insteon.com/products/hub-pro-refurbished)  These are a Beaglebone black board connected to a serial PLM inside the box.  You can install Homebridge and the plugin directly on the Hub and it works like a charm.  Downside is that you'd have to redo links, but a relatively inexpensive solution.

**Reply by mtwunch** [reply]
@kuestess This is quite ironic!  It does present an inexpensive solution.  Thank you for posting this!

**Comment by mtwunch** [comment]
@kuestess I now have Homebridge running on a Pi.  I am getting an error that appears to be a syntax error, but I cannot resolve it.  The log says TypeError.  Here is a log snippet:
[11/27/2024, 3:46:49 PM] [InsteonUI] Initializing Insteon UI 
[11/27/2024, 3:46:49 PM] [InsteonUI] Reading config from /var/lib/homebridge/config.json
[11/27/2024, 3:46:49 PM] [InsteonUI] Found 2 platform(s) in config
[11/27/2024, 3:46:49 PM] [InsteonUI] Creating new insteon.json
[11/27/2024, 3:46:49 PM] [InsteonUI] No devices in config
[11/27/2024, 3:46:49 PM] [InsteonUI] Done with InsteonUI init
[27/11/2024, 15:46:49] [InsteonLocal] Found 3 devices in config
[27/11/2024, 15:46:49] [InsteonLocal] Adding new accessory: Front Door
[27/11/2024, 15:46:49] [InsteonLocal] Adding new accessory: MR2
TypeError: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
    at Hash.update (node:internal/crypto/hash:115:11)
    at Object.generate (/var/lib/homebridge/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/uuid.ts:11:11)
    at InsteonLocalPlatform.discoverDevices (/var/lib/homebridge/node_modules/homebridge-platform-insteonlocal/src/InsteonLocalPlatform.ts:659:38)
    at HomebridgeAPI.<anonymous> (/var/lib/homebridge/node_modules/homebridge-platform-insteonlocal/src/InsteonLocalPlatform.ts:116:12)
    at HomebridgeAPI.emit (node:events:519:28)
    at HomebridgeAPI.signalFinished (/var/lib/homebridge/node_modules/homebridge/src/api.ts:275:10)
    at ChildBridgeFork.startBridge (/var/lib/homebridge/node_modules/homebridge/src/childBridgeFork.ts:194:14)
[27/11/2024, 15:46:49] [InsteonLocal] Child bridge process ended
[27/11/2024, 15:46:49] [InsteonLocal] Process Ended. Code: 1, Signal: null

And a config snippet:
 {
            "name": "InsteonLocal",
            "host": "/dev/ttyUSB0",
            "port": "25105",
            "model": "PLM",
            "refresh": "0",
            "server_port": "3001",
            "keepAlive": "0",
            "devices": [
                {
                    "name": "Front Door",
                    "deviceID": "13C16A",
                    "dimmable": "no",
                    "deviceType": "keypad",
                    "keypadbtn": "A"
                },
                {
                    "dimmable": "true",
                    "deviceType": "dimmer"
                }
            ],
            "platform": "InsteonLocal",
            "_bridge": {
                "username": "0E:C1:7C:5C:A2:DB",
                "port": 31456
            }
        }

It is probably something simple that I am overlooking....

**Reply by kuestess** [reply]
@mtwunch You've got an extra, invalid device definition in your config.  Just remove the 
```
{
"dimmable": "true",
"deviceType": "dimmer"
}
```
lines from your config and it should work.

**Reply by mtwunch** [reply]
@kuestess Okay...I will give that a try.  Thank you!
